packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js

```diff
@@ -213,8 +254,17 @@ const HomePage = () => {
     _page: generatePageFromStart(start, limit),
   };
 
-  const areAllCheckboxesSelected = data.length === dataToDelete.length;
-  const hasSomeCheckboxSelected = dataToDelete.length > 0;
+  const paginationCount = data.length < limit ? data.length : dataCount;
+
+  const hasSomeCheckboxSelected = data.some(item =>
+    dataToDelete.find(itemToDelete => item.id === itemToDelete.id)
+  );
+
+  const areAllCheckboxesSelected =
```

Comment:
This won't work if you select all the items from page 1 and go to page 2

Reply:
We want each `select all` checkbox independent. If I select all items from the first page, the select will be checked. And if i go to page 2 and nothing is selected, the checkbox is unchecked. 

As it i can select all items from the page 1, go to another, select all and then unselect all items, come back to the page 1 and all the page 1 items are still selected. I wish it's understandable, let me know if it's not!